lib/spack/spack/package.py

```diff
@@ -973,6 +973,79 @@ def url_for_version(self, version):
         return spack.url.substitute_version(
             default_url, self.url_version(version))
 
+    def urls_for_version(self, version):
```

Comment:
If this were a generator, could `url_for_version` just pick the first value yielded? I'm noticing some logical overlap between this function and `url_for_version`, and I also find it confusing to have functions named `url_for_version` and `urls_for_version`: perhaps if this were called `all_urls_for_version` that would help distinguish it from `url_for_version`.

Reply:
Probably, or at least I could make that possible.

Reply:
This turns out to be harder than I would have thought.  Since `url_for_version` is the customization point for packages, I need to call it in `all_urls_for_versions`.  I tried to detect when it's customized and only call it then, but that is proving to be unreliable.  I'll be happy to do the rename, but getting the singular to use the multiple is not straightforward.